### DISCUSSION TITLE
More Structured Progression

### DISCUSSION THREAD

**Discussion by Hecter94** [endless-sky/endless-sky#9193] [Ideas]
I have recently completed a playthrough of Endless Sky on continuous version https://github.com/endless-sky/endless-sky/commit/d21ee81fd2343439aefc9f1ade3b5db527defaaf
Throughout this playthrough, I avoided sequence breaking or using metaknowledge, meaning I went through what I believe to be the intended "progression" path.

Following the intended progression path worked great until the end of the Free Worlds questline.
As soon as I finished Free Worlds, got my Jump Drive, and was interested to begin exploring on my own, like maybe going to check out the Korath or the hidden systems to the south of FW, which I had spotted previously but wasn't able to get to, Hai Reveal sent me straight to Hai space which I had not discovered yet; going to Hai space then triggered both Deep Archeology and Deep Research to continue, Deep Research then led me straight into both Gegno and Remnant. HR quickly led me to the Unfettered, leading me to Wanderer shortly after.

Meaning as soon as I completed FW, I suddenly had about four new races and five new storylines, all trying to start at the same time, all of which were vying for my attention, leaving me unable to appreciate any of them as I bounced between various regions of space doing all of these storylines in tandem.
As you can imagine, this was slightly overwhelming, with so much new stuff all trying to happen simultaneously, and a newer player that doesn't understand any of this would have been even more overwhelmed by it. 

As such, it would be a good idea to structure the progression chain of ES a bit more than it is currently, at least in the very early game. 
To that end, I believe that an ideal progression chain for ES is something like this:
- The player begins the game and completes the James tutorial mission chain, introducing them to some basic concepts 
- The player, with the starting debt as motivation, begins to explore human space, doing jobs and the odd mission, such as Pact recon, Pookie, TMBR, Bactrian questline, etc.
- The player starts questlines such as Deep Research and Deep Archeology but can't complete them.
- The player, after paying off their starting debt and upgrading their ship a bit, discovers Hai space.
- The player explores Hai space, is introduced to the faction, and completes some jobs/minor missions for them.
- The player also discovers Unfettered Hai at this time but is blocked from progression due to the lack of a jump drive.
- The player is now able to return to human space and, due to discovering the Hai, can progress in Deep Archeology and Deep Research
- The player begins exploring the Ember Waste due to Deep Research. However, a soft block prevents them from easily discovering Gegno or Remnant.
- The player completes Deep Research and is blocked from completing Deep Archeology due to lacking a Jump Drive.
- The player can now begin the Main Plot and should be able to complete it without any further blocks or issues.
- The player obtains a Jump Drive after completing the Main Plot and can now explore freely. 

At this point, it is important to note that immediately after obtaining a jump drive is *not* the time to hook the player into any new storylines. We've already set up a variety of hooks by telling them about the Korath and their basic location, by setting up a soft block in the Ember Waste which they can now bypass, by blocking their progression with both Unfettered Hai and Deep Archeology until they're able to obtain Jump Drives, which they now can do. 

At this stage of the game, the player should *not* be pulled into any new storylines; they should be free to choose which of the previous hooks we've set up that they wish to follow now. 
A player may choose to:
- Explore the Coalition systems they may have spotted previously but were unable to get to, leading them to discover the Coalition and doing the Coalition storyline
- Bypass the soft block they encountered previously in the Ember Waste, allowing them to discover Gegno and/or Remnant and start their storyline(s)
- Go check out the Korath that they heard about from the Syndicate, potentially allowing them access to more Jump Drives and allowing them to start/continue Deep Archeology or Unfettered Hai
- Return to Hai Space at any point and trigger Hai Reveal to start

The important thing to note is that we've now had a very structured early game that introduces the player to the galaxy at large and to a couple of different factions and, most importantly, sets up a variety of hooks that can't be completed immediately. 
We've then given them a Jump Drive and opened up the galaxy, giving them the freedom to choose which of those hooks they wish to pursue and allowing them to complete all of the remaining content in essentially whatever order they choose, *without* immediately trying to drag them into all of those different storylines. 

A structured beginning sets up concepts and intrigue when the player is released from that structure and given free rein to explore how they see fit. 
Whereas right now, we have a very loose beginning where the player can easily get lost or stumble into things before they're ready to and then try to pull them *into* a structured narrative at the moment we should give them freedom. 

To accomplish this, we would need to do five things. 
1) Apply more "soft blocks" in the early game. Obstacles that can be bypassed but otherwise helps to guide new players away from the content they need more time to be ready for and towards the ideal progression chain. 
2) Use the structured progression in the early game to tutor the player and set up hooks for them to follow later once they are given freedom
3) Give the player freedom after they have obtained a Jump Drive, don't try to pull them into any new storylines at this point, and allow them to discover things organically.
4) Allow the mid-game content to be completed in any order driven solely by the player's desire for exploration
5) Use end-game content to tie the mid-game storylines together and bring everything to a satisfying conclusion.

If we can accomplish these five points and set up Endless Sky with a structured beginning that branches out into complete freedom, we can create a much more pleasant experience, especially for newer players. 
However, I leave the specifics as to how those five points can be accomplished up to discussion.

**Comment by Hurleveur** [comment]
What kind of soft locks are you thinking of?

For the remnants we already have the keystone lack of passive ramscoop and the ion storms (tho we could add some hostile creatures ig)
As for HR I was under the impression it took a long time to trigger now, with the reworked prologue? (And tbf any non deadline mission ca wait forever, mb we just need to be clear about it here)

Also I dont see why we should block deep missions as they provide smth to look forward to, or some intrigue at least

**Comment by Quantumshark** [comment]
You make a couple of points I agree with here, but I don't agree with your concept for a solution or with several other things you've said.

>Meaning as soon as I completed FW, I suddenly had about four new races and five new storylines, all trying to start at the same time, all of which were vying for my attention, leaving me unable to appreciate any of them as I bounced between various regions of space doing all of these storylines in tandem.
As you can imagine, this was slightly overwhelming, with so much new stuff all trying to happen simultaneously, and a newer player that doesn't understand any of this would have been even more overwhelmed by it.

This part I agree is a concern. Once the campaign is over, we should not be trying to push the player around. They should be left to explore freely, and we should be open to the fact that they could find and begin any one of several storylines, or postpone the storylines and start mapping out the galaxy first, or anything else. *There should not be an intended sequence of progression after the end of the main campaign.*

>The player starts questlines such as Deep Research and Deep Archeology but can't complete them.

This is an important point. Yes, the player may not initially be able to complete these missions. But, crucially, there isn't anything saying "come back here once X" - rather, both seem to end, and then re-emerge unexpectedly later. I think this false end and then surprising reappearance works a lot better than trying to leave an "I need to go back to this place after I do this thing I don't know how to do" in the back of the player's mind (or, worse, their mission log).

> The player, after paying off their starting debt and upgrading their ship a bit, discovers Hai space.

No. This one I am fundamentally opposed to. Hai space should be something that the player *might* discover before the campaign, not something they *will* discover. Both "the player discovers the Hai before the main campaign" and "the player does not discover the Hai until after the main campaign" should be valid progression paths - this is important to making the discovery of the Hai feel like an accomplishment for the player.

> The player also discovers Unfettered Hai at this time but is blocked from progression due to the lack of a jump drive.

And this does the opposite of what I mentioned about the deep archaeology missions, leaving a note to linger in the back of the player's mind, as a distraction to which a lot of people are going to respond, naturally, by thinking "OK, so the next thing I need to do is get a jump drive", and either sequence breaking or deliberately ignoring anything that doesn't seem JD-related with the intent of coming back to it later. But also, I feel like it's best if the player first learns about the Jump Drive from the main campaign, rather than hearing about it from the Unfettered or finding one for themself.

>The player is now able to return to human space and, due to discovering the Hai, can progress in Deep Archeology and Deep Research
The player begins exploring the Ember Waste due to Deep Research. However, a soft block prevents them from easily discovering Gegno or Remnant.
The player completes Deep Research and is blocked from completing Deep Archeology due to lacking a Jump Drive.
The player can now begin the Main Plot and should be able to complete it without any further blocks or issues.

Two things here. Firstly, I dislike at a fundamental level this idea that the sequence of events should be fixed. I think it really adds something that two people can play through the game, find different mission chains and secrets, and have completely different experiences of the game. It's really enjoyable when you're new to the game to talk to other players and compare your experiences, and you talk about different things that one of you found and the other didn't.
Secondly, I don't like this idea of these 'blocks'. I don't think seeing a barrier that says "come back later once you've done X" is good game design, because it means the game world doesn't feel like a real world, it feels like a series of paths that are directing the player down a set sequence of events.

**Reply by Hecter94** [reply]
To be clear, when I say "ideal progression chain," I'm referring to what I think would lead to the best experience for the player. Not the chain that every player will always be forced to do in that exact sequence.
You can keep the idea of open progression but still provide guidance toward the specific route that will lead to the best experience without completely blocking off all other paths. 

Hence the idea of a soft block. Something that functions as a light barrier, which can be bypassed with meta knowledge or by a determined explorer, but functions to help steer players towards an intended progression path without making that the only path available.

**Reply by MasterOfGrey** [reply]
I find myself generally in agreement with Quantumshark. Especially as now the push towards HR is so much more stretched out than it was before.
Including that idea that it'd be good if the UHai didn't ask you about jump drives until you actually have one already.

**Reply by Zitchas** [reply]
I agree with Quantum shark. The Ember Wastes are designed to be a soft block already. If need be they can be tweaked, but I see no need to add a new block in here.

I very much like not having an intended path, and the more blocks are out in place to shape the path, the harder I as a player will push back and try to get outside them. This is why I generally do everything possible *other* than the main campaign whenever possible. 

I generally feel that the Emerald Sword ought to be relaxed a bit, so it can be completed and acquired sufficiently early to actually be useful.

**Comment by Arachi-Lover** [comment]
I really dislike the idea of having the whole game structured on rails.
There are certain parts where of course locking X behind Y is a better approach (ie locking Wanderers behind the human campaign, as that is dependent on said campaign given characters that are involved and events that take place), but in general? No.

**Reply by DarcyManoel** [reply]
Couldn't agree with Arachi more, freedom of progression is paramount.

**Comment by Hecter94** [comment]
So I'm seeing the same couple of misconceptions being brought up in most of the replies, which makes it likely that I could have communicated my point more effectively. For that, I apologize. 

To clarify a couple of things then:
The purpose of this post is *not* to say that progression through ES should in any way be "fixed" or "on rails." That sentiment is not expressed anywhere. I'm sorry if I didn't make that clear.
The purpose of this post is, in a nutshell, that we should have *some* idea of what the "ideal" progression through ES is and apply structure to the game's story, mechanics, and world to guide players toward that ideal progression and away from areas that we think would be detrimental to the play experience of a first-time player.

To reiterate that point, this is *not* about fixing the progression of ES so that only one route is possible. This is about applying structure to guide players onto the route we feel gives the best experience _without_ making that the only route possible. 
It is very much *expected* that players will likely deviate from the "ideal" route, which is normal, especially on subsequent playthroughs. 

The point is to apply structure to that progression so that while many routes are possible, players, especially ones that are new to the game and don't know what they're doing, are subtly guided onto the route we feel is the "ideal" one.
Which, to reiterate, the "ideal" route is *not the only route*. It is the one we feel is the best among the many possible. 

I will give examples from other games to communicate this point better. Light spoilers for the following:

<details>
<summary> Fallout: New Vegas</summary>

When players start a new game in New Vegas, they are given a short tutorial and an objective. In this case, the objective is ultimately to "get to New Vegas," as that is the trigger point that will push the story forward. 

To illustrate what I'm talking about, I am going to provide an annotated map and will be referring back to the annotations as I continue:
![image](https://github.com/endless-sky/endless-sky/assets/15916854/e6bf30d5-d553-49c1-a89e-0d1cd1777627)


The Green "O" is the Player's starting point, and the Red "O" is the eventual destination they are trying to get to. 
I have marked the shortest path(s) from the Player's current position toward their eventual destination with a blue line.

However, most players on their first playthrough won't take the shortest path; they will take a much longer path I marked in green, with some likely detours marked in Yellow. 

The reason *why* players take the Green route instead of the much shorter Blue route is two-fold:
1) The blue routes have "soft blocks" placed directly on them, denoted by the red "X"; those "soft blocks" would be very difficult to bypass without a solid understanding of the game's mechanics. It is unlikely that a new player would be able to get past those "soft blocks" unless they are very determined to do so and thus take the easier of the two options and progress down the Green route.
2) The game's story makes it clear to the Player that the blue route is dangerous and points them toward the Green route instead, along with the diversions denoted in Yellow.

If a player then does choose to take the green route instead of the blue route, the game uses the roundabout path to better introduce the player to the world they're in; it uses the extra time to introduce them to factions, story elements, and key characters, and sets up some hooks that the player will be interested in returning to once they've completed their current objective.
Effectively, for a new player, following the green route, while optional, gives a much better experience than following the blue route. Therefore the game is structured in such a way as to encourage the Player to *choose* the green route by placing soft blocks in their path that can only be easily overcome with a decent knowledge of the game's mechanics. 

So to be clear, at the start of the game in New Vegas, while the Player *does* have complete freedom in where they can choose to go and can take any of the possible paths leading out of their starting area, the game's structure is designed as such, with guidance and with soft blocks to steer the player towards the route that gives them the best experience with the game, while still allowing players complete freedom over where they choose to go, and with more experienced players very often taking the blue routes on subsequent playthroughs.

It's also worth noting that the previous "main" titles in this series, Fallout 1, 2, and 3, all did similar things. 
Fallout 1 placed its "tutorial town" directly between your starting location and your objective, meaning any player that tries to go straight to the objective will wind up in the tutorial town instead.
Fallout 1 & 2 surround the starting area with tough but surmountable encounters everywhere other than the intended route, again encouraging players to follow that route.
Fallout 3 Starts the player facing directly toward the "tutorial town" and uses some clever scenery tricks to make the "intended" route look much more visually interesting, assuming players will go towards what looks interesting. 

In all of these cases, you are not "railroaded" onto the intended path; you are, in fact, in all four games mentioned, able to run straight from your starting area to an endgame area to skip most of the story if you so choose. Still, through clever design and soft blocks, they encourage the player to "choose" the route that leads to the best experience for a new player. 
</details>

<details>
<summary> Dark Souls 1</summary>

When a player starts a new game in Dark Souls, they complete a short tutorial section and are deposited into the open world with virtually zero direction on where to go. 
From a player's starting point, they have quite a few different routes they could choose to travel:
![image](https://github.com/endless-sky/endless-sky/assets/15916854/a925849d-1ff0-4ab2-953a-c4c5275663a5)

However, of those four possible routes, only of them (in green) will give a good play experience; the other three routes will either block the player off at some point or just be very frustrating as they will come up against foes that they won't have the skill or knowledge to be able to defeat. While the green route pits the player against reasonably challenging enemies and provides access to several useful early-game items and NPCs.

As such, to ensure that more than 25% of new players will randomly pick the best route forwards, the game does two clever things. 
1) It places "soft blocks" along all unintended routes. Whether those are difficult enemies, a locked door that can't be easily bypassed, or a tough NPC fight that needs to be defeated to progress.
2) As soon as you start in this new area, you're immediately left facing directly toward where you are expected to go, and the expected path also has the most interesting-looking visuals:
![image](https://github.com/endless-sky/endless-sky/assets/15916854/31a2ba65-9192-410b-ae10-465780a12475)

So while you are, again, not *forced* to go this way, the game is structured so that it will *encourage* you to take this route, leading to the best play experience. 
You are not railroaded into going this way, you can go in other directions as well, and more experienced players very often will. Still, you are *encouraged* to choose this specific route forward due to guidance in the "right" direction and soft blocks in the "wrong" direction.

</details>

<details>
<summary>Elden Ring</summary>

To give a more recent example, Elden Ring employs many of the same tricks while being a much more "open-world" game. When you start a new game in Elden Ring, the vast majority of the map is open to you, and there are several different directions you *can* go in.
![image](https://github.com/endless-sky/endless-sky/assets/15916854/dc3f5f9d-dfc8-4b80-a4d6-770b1e2de8e2)

Almost everything in the Blue Circle is available to new players when they start the game without fighting a single enemy. They can run anywhere within that blue circle straight from the starting zone, and I've drawn several such paths in yellow. 

However, again, while Elden Ring is a bit more relaxed in this regard, there is an "intended" progression order which is what leads to the "best experience" I am not going to draw it this time as it involves teleporting and fast traveling and backtracking and as such is a bit too complex to display on a 2d map, but suffice to say, it is *not* any of the yellow routes. 

However, if all of those routes truly were open to a brand-new player, it would be extremely overwhelming for them, and the chances of them stumbling through things in an order that gives a very poor experience would be quite high, so the game does several very clever things to encourage players to follow the "intended route."

For example, several of those "yellow" routes include hidden passes or require a good bit of exploration to access them; some may be as simple as a hidden pathway built into the side of a mountain or a grand lift that needs a hidden key to open, or a treasure chest at the bottom of a dungeon that teleports you to a new area.
Most of these are things a newer player will likely stumble upon if they are a determined explorer. 

In addition to this, the game quite literally points you toward the "intended route." If we take a closer look at the map:
![image](https://github.com/endless-sky/endless-sky/assets/15916854/0411764b-c021-4c78-8129-223c72fbb1f9)

These yellow arrows very much *literally* point you toward the intended route. You are not *forced* to follow them, and there are other routes you can take, but you are very much guided toward the route the developers feel will give the best experience. 

So while, again, the player is not forced to go the "intended" way, they are free to pick whatever route they choose, the fact that they are guided towards the intended route and the paths that they need to take to go on different routes are somewhat hidden or difficult to access allows the game to offer a lot of freedom of choice still and still allow a lot of different options for where a player can choose to go, without it being so lacking in structure that players don't get a good experience. 

</details>

Virtually every "Bethesda RPG" and "FromSoft Souls-Like" does this excellently. I could talk about how Skyrim, Oblivion, Dark Souls 2, 3, Bloodborne, etc., all do very similar things to great effect, but at that point, I'd largely just be repeating myself. 
In conclusion, virtually every game in those categories is an excellent example of how you can do an open world with complete freedom while still including enough structure to guide players down the route you want them to go without railroading them into that route. 

Hopefully, those examples will show my point. 
I am *not* advocating, in any way, to make progression in ES into a completely "linear" or "fixed" experience; I think it's great that we allow different progression paths.
However, I do think that we should have an idea of what we feel the "ideal" path is, or what order we "intend" for a newer player to progress through the game in, and then structure the game in such a way, either through direct guidance or soft blocks in key areas to encourage them along the path that gives them the best experience. *Especially* if they are a newer player that needs to be introduced to concepts and eased into the game before they should be stumbling into later-game areas. 

To reiterate, this is not about ensuring that only one progression path is available. This is about adding structure to the game to encourage a certain intended path for players that still need the slower introduction to concepts and mechanics within the game and could benefit from being introduced to these things in the "correct" order. 

I gave my opinion on the ideal progression path and where we could put some soft blocks to encourage that specific progression path. You may disagree with what I feel is ideal or where/what those blocks should be, and that is 100% okay; the point of this post is not that we should railroad players down the specific path that I felt was ideal, but rather that *we* should have some idea of what the ideal progression route is, and then look at ways of subtly guiding newer players down that path without completely blocking off all other paths. 

This means that there are, effectively, three areas that warrant further discussion. 
1) Should ES follow the examples I gave above and include structure encouraging players towards an intended path?
2) If yes, what progression path would give a new player the best experience? (I gave my opinion above, feel free to disagree)
3) What sort of guidance/soft blocks could we build into the progression to encourage following that progression path? (I gave my opinion above, feel free to disagree)

Hopefully, I've clarified my position better and we can now discuss those points specifically.

**Comment by Arachi-Lover** [comment]
I don't think those are very good examples.

NV in particular was extremely frustrating to me due to how absurdly on rails it seemed; the fact that the paths you highlighted are **the paths** made me groan and think "really?" dozens and dozens of times on multiple playthroughs.
The reason being, NV is absolutely terrible at providing me with believable natural barriers to justify why I can't just cut a straight line from Goodsprings to Vegas. There are some shy, not steep at all hills on the way, and that's it. But I keep hitting invisible walls because I'm supposed to believe I can't climb that. Neat.

From Software games on the other hand, such as DS and ER like you highlighted, have way over-the-top barriers that are pretty clearly not passable just by basic walking or climbing. However, these games also suffer from extremely contrived blockages and gateways.
Sen's Fortress' whole existence is a gigantic sore thumb, the Lordvessel magically destroying some walls because uhhh something something you weren't supposed to kill Seath or Nito yet.

Both types of irksome issues with the barriers and mechanisms put in place in those games to try and steer the player into a certain "intended" route or ideal play order are very immersion breaking when it comes to the story, and it becomes clear the developers needed a way to extend the gameplay time you were on a specific task or objective, but couldn't find means that did that that could be justified within universe, so we end-up with very easily climbable hills that are stacked with invisible walls, and random castles and magically disappearing walls.

Given the story-driven element of ES, I really don't think we should look at those games as good examples of softly steering gameplay into a certain way.

**Comment by Zitchas** [comment]
A few thoughts:

1. While there may have been some justifiation to gating the Emerald Sword to post-FW originally, at this point I don't think it is necessary.
2. Most of the timing issues can be solved by having more interesting stuff to do in the core human space. And by that I mean have enough content that people are running into new unique events, missions, or mission strings every 4-8 landings for about a decade of puttering around human space. Interesting stuff to find and do is the guaranteed number one best way to keep people from haring straight off into the great beyond in search of aliens the instant they have the capacity to do so. Every company mentioned in the game should have at least a mini mission string; potentially even a major mission string or mini-campaign of their own. Implementing a bounty-hunter guild and unlocking progression through more challenging bounties would both add a lot of story potential, gate bounty missions a bit to keep people from jumping head first into stuff they can't handle, and make things more interesting. Likewise, I think every single iteration of EV had a courier company of some kind with a series of missions that unlocked better jobs, typically small, tight deadline, high value jobs that, given the hyperspace time passage increasing for bigger ships in those games, were jobs that were only completeable by small fast couriers (baring tricks and other unlockables, anyway). And it should be mentioned that the overwhelmingly vast majority of these should be internal missions, stuff dealing with human problems and situations, bouncing around human space. Not pushing the player to explore unknown territories. Some of that is great, but most missions should be doing stuff at home.

tldr: Give people more stuff that is interesting to do that is purely internal to human space, and people will spend more time in human space. Caveat: It has to be *interesting*, and preferably with a decent amount of *meaningful* stuff. Including unlocks.

**Reply by junetried** [reply]
I agree that Human space needs more content to keep the player interested for longer, otherwise they can easily wander off into missions they aren't equipped for. That's exactly what I did on my first (blind) game; I entered the Free Worlds line because nothing else seemed to appear, with my flagship being a Bounder. It might've saved a lot of frustrations if I found more things to do before the Free Worlds campaign started.

I also think the campaign missions with no deadline need to be less narratively "urgent" - it doesn't make a lot of narrative sense to agree to help the Free Worlds scan Republic ships or such, and then take a month-or-two long detour because you found another thing to do, even if nothing actually stops you. I think the Wanderer and especially Remnant story lines are better about making it clear when you can go do other things, and I think that's a good thing.

**Comment by mOctave** [comment]
There are good games that are on rails, but when the rails run out the player gets bored.
Endless Sky is story-based, not a story.
I think we all agree on that.

However, creating a structured story is _also_ not the point of ES. We shouldn't have an "ideal" path to completion, because sooner or later, someone is going to not follow that and then creatively break the game in an unexpected way (ie, me doing the Bactrian and Deep Archaeology questlines at the same time). When there's just a rough structure in place, it works okay, and maybe you get a few laughs out of it along with a lesson to not play for both teams at once. When everything else is structured, however, having some missions inexplicably overlap in difficult ways is much harder to handle.

I think it's okay if there's four or five factions vying for the player's attention. They can choose one and come back to the rest later.

Personally, one of the things I enjoy so much about ES is how there's no normal way to do things. Eventually, we'll probably have ten or fifteen routes through the war, with you working for three sides, making one or two big decisions and a whole bunch of small ones. Meanwhile, you can go complete the story in nearly any order, build up an enormous fleet, collect one-of-a-kind vessels, hack your save and play with the mechanics, or install plugins that make the game do things it was never designed to do. That's the beauty of ES. Nothing is intentionally discouraged save what the player's conscience deems immoral. If we change that, I think we risk losing the spirit of the game.
